src/modules/product/index.ts

```diff
@@ -75,4 +75,88 @@ async function fetchProducts(params: any): Promise<any | Response> {
   return response;
 }
 
-export { fetchProducts }
+async function fetchProductsAsGroups(params: any): Promise<any | Response> {
```

Comment:
We should improve this. This method covers diverse use cases. We should have specific methods based upon the use case link there could a method that returns variants grouped by virtual products. We could use this as a generic method and  have wrapper calling it for specific use case